tools/logicalimager/RegistryAnalyzer.cpp

```diff
@@ -191,10 +192,13 @@ USER_ADMIN_PRIV::Enum samUserTypeToAdminPriv(uint32_t& acctType) {
 
 int RegistryAnalyzer::analyzeSAMUsers() const {
     std::map<std::wstring, FILETIME> acctCreationDateMap;
-    RegFileInfo *aRegFile = RegistryLoader::getInstance().getSAMHive();
+    RegistryLoader *registryLoader = new RegistryLoader();
+
+    RegFileInfo *aRegFile = registryLoader->getSAMHive();
     if (aRegFile == NULL) {
         fprintf(m_outputFile, "SAM HIVE not found\n");
         fclose(m_outputFile);
+        delete registryLoader;
```

Comment:
This seems strange. Why make a new RegistryLoader if the one we have has already loaded the hive data?

Reply:
I want to load hive(s) from different drive(s) independently.
In the old code you may load multiple hives into a single instance.

Reply:
This is so that registry hive(s) in different drive(s) are independent of each other.
The old code was a singleton.